lib/block-supports/border.php

```diff
@@ -6,26 +6,36 @@
  */
 
 /**
- * Registers the style attribute used by the border feature if needed for block types that
- * support borders.
+ * Registers the style attribute used by the border feature if needed for block
+ * types that support borders.
  *
  * @param WP_Block_Type $block_type Block Type.
  */
 function gutenberg_register_border_support( $block_type ) {
-	// Determine border related features supported.
-	// Border width, style etc can be added in the future.
-	$has_border_radius_support = gutenberg_block_has_support( $block_type, array( '__experimentalBorder', 'radius' ), false );
+	// Determine if any border related features are supported.
+	$has_border_color_support = gutenberg_block_has_support( $block_type, array( '__experimentalBorder', 'color' ) );
+	$has_border_support       =
```

Comment:
It looks like we should check for `$has_border_support = gutenberg_block_has_support( $block_type, array( '__experimentalBorder' ) );` instead of the individual properties.

Otherwise, we may run into this situation:

- a block.json has `support.__experimentalBorder: true`.
- the client will [accept](https://github.com/WordPress/gutenberg/blob/da9b4ae772801c311112cf56fa35d1525490b3f5/packages/block-editor/src/hooks/style.js#L33-L33) that as support
- the server will not because it checks for the individual properties

Reply:
Thanks for catching this one. I've fixed that oversight.